tracee/tracee_test.go

```diff
@@ -120,16 +120,40 @@ func TestReadArgFromBuff(t *testing.T) {
 			},
 			expectedArg: map[string]string(map[string]string{"sa_family": "AF_INET", "sin_addr": "255.255.255.255", "sin_port": "65535"}),
 		},
+		{
+			name: "sockAddrT - AF_UNIX",
+			input: []byte{12, //sockAddrT
+				0,    // Dummy tag
+				1, 0, //sa_family=AF_UNIX
+				47, 116, 109, 112, 47, 115, 111, 99, 107, 101, 116, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 101, 110, 0, 0, 0, // sun_path=/tmp/socket
+			},
+			expectedArg: map[string]string{"sa_family": "AF_UNIX", "sun_path": "/tmp/socket"},
+		},
 		{
 			name:          "unknown",
 			input:         []byte{0xDE, 0xAD, 0xBE, 0xEF},
 			expectedError: errors.New("error unknown arg type 222"),
 		},
+		{
+			name: "strT too big",
+			input: []byte{10, //strT
+				0,          // Dummy tag
+				0, 0, 0, 1, //len=16777216
+			},
+			expectedError: errors.New("string size too big: 16777216"),
+		},
 	}
 
 	for _, tc := range testCases {
-		_, actual, err := readArgFromBuff(bytes.NewReader(tc.input))
+		b := bytes.NewReader(tc.input)
+		_, actual, err := readArgFromBuff(b)
 		assert.Equal(t, tc.expectedError, err, tc.name)
 		assert.Equal(t, tc.expectedArg, actual, tc.name)
+
+		// TODO: the following fails for:
+		// strArrT
+		// unknown
+		// sockAddrT - AF_UNIX
+		assert.Empty(t, b.Len(), tc.name) // passed in buffer should be emptied out
```

Comment:
In the case of unknown, we can't consume the event from the buffer since we don't know the event's size. (we should stop tracee in this case). so I don't think we should assert for emptiness in this case

Reply:
Does that happen today? (The stoppage of Tracee that is) I just see all errors are collected and printed via a handler.

Reply:
Anyways - that's tangential to this PR, we can take it offline.